Feature/initial module

jamesjwarren opened the pull request:
This PR adds the SkrollrService and snSkorllr directive which load and then initialise the skrollr animation elements. The directive refreshes skrollr when an element with the directive is loaded.

jamesjwarren committed:
initial SkrollrService and snSkrollr directive

jamesjwarren committed:
update README

edoparearyee commented on app/js/SkrollrService.js:
```diff
@@ -0,0 +1,63 @@
+        scriptTag.type = "text/javascript";
+        scriptTag.async = true;
+        scriptTag.src = "components/skrollr/dist/skrollr.min.js";
+
```
Why inject the skrollr library tag using javascript? This presumes the person using this directive has the same directory structure as us e.g. has the skrollr lib in `app/components/`. Also this won't work, even for us, when all the javascript is concatenated and/or minified.

jamesjwarren replied:
Yeh I was concerned about this too. The reason I have used this method is we need to initialise skrollr and directives after it has loaded. 

Just including the script in the skrollr is never defined before it is used.

edoparearyee replied:
https://github.com/Prinzhorn/skrollr#lets-get-serious
You just need to wait till the document loads before initialising your directives. You can use the `window.onload` event to be sure skrollr is defined. It will only run once all assets e.g. css files, javascript files have been loaded.

jamesjwarren replied:
Ahh for some reason I had assumed angular ran within window.load anyway so neglected to think of it. Does the trick, cheers.

jamesjwarren committed:
remove loading from SkrollrService

jamesjwarren committed:
refactor skrollr initialisation

jamesjwarren committed:
remove initialisation from refresh directive

jamesjwarren committed:
update README

jamesjwarren commented:
I've refactored the skrollr initialisation and added an snSkrollrInit directive which allows one to pass in init options like this:

``` html
<body sn-skrollr-init="{ smoothScrolling: true }">
...
</body>
```

edoparearyee commented:
Instead of using a directive for this why not use a provider?
https://docs.angularjs.org/guide/providers

For example in routeProvider you configure the ngRoute service before the app is run. Providers are perfect for reusable services that require a one time config before the app is run.

jamesjwarren committed:
replace initialisation directive with provider

jamesjwarren committed:
update README with provider config usage

jamesjwarren commented:
I have now replaced the initialisation directive and service with a provider which enables skrollr to be configured in the angular config phase and initialised within `app.run()`.

edoparearyee commented on app/js/skrollrProvider.js:
```diff
@@ -0,0 +1,46 @@
+
+    var _this = this;
+
+    this.init = {};
```
would it make more sense to call this object `config` instead of `init`?

jamesjwarren replied:
agreed :+1:

jamesjwarren committed:
rename init object to config

jamesjwarren commented:
We probably don't need to use the promise library anymore either.

jamesjwarren committed:
remove $q - no longer required

edoparearyee commented:
looks good, i think you just need to fix the build and it will be ready for a release.

Also you may want to start think about writing test to improve code coverage, it should be fairly straight forward though, since this is very small.

(actually ignore that, i meant tests for the thisissoon-frontend)

jamesjwarren committed:
unit test for snSkrollrProvider

jamesjwarren committed:
combine module into single file

jamesjwarren committed:
unit tests

edoparearyee commented:
Looks good :+1: 

coveralls seems to be slow, will merge and release once it fixes itself
